django_loci/base/admin.py

```diff
@@ -230,7 +230,10 @@ def _get_location_instance(self):
         location.name = data.get('name') or location.name
         location.address = data.get('address') or location.address
         location.geometry = data.get('geometry') or location.geometry
-        return location
+        if not location.is_mobile and not location.geometry:
+            raise ValidationError(_('Geometry can\'t be null'))
```

Comment:
move this logic to the base model, that's where it belongs, the admin is just a view, the main logic (also known as business logic) goes into the model, at least in django apps this is how we do it

Reply:
Okay.. @nemesisdesign i suppose i have to put it here https://github.com/openwisp/django-loci/blob/1c5ed122f526e3ba7f88b27cb67f4010d7493e95/django_loci/base/models.py#L20 right? If yes then do i have to make a separate function to check this?